packages/sdk/src/evm/lib/static-batch-rpc.ts

```diff
@@ -66,6 +66,10 @@ export class StaticJsonRpcBatchProvider extends providers.StaticJsonRpcProvider
         // on whether it was a success or error
         batch.forEach((inflightRequest_, index) => {
           const payload = result[index];
+          // if there is no payload, reject the request
+          if (!payload) {
+            inflightRequest_.reject(new Error("No response for request"));
```

Comment:
Should this be an if/else statement with `if(payload.error)` ? I think after rejecting we don't want to continue the execution

Reply:
good catch, will fix

Reply:
actually no, we want to process each inflight request separately

Reply:
Yeah, bit if there's no payload we should continue to the next request, if I understand this correctly

Reply:
and we do, we're in a forEach loop there. we're resolving / rejecting n promises

Reply:
added an explicit `else` case